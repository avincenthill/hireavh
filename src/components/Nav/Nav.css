.nav {
  background-color: var(--c-4);
  position: absolute;
}

.nav-bottom {
  bottom: 0;
  right: 0;
  margin-bottom: var(--s-s);
  border-radius: var(--s-xs) 0 0 0;
}

.nav-top {
  top: 0;
  left: 0;
  margin-top: var(--s-s);
  border-radius: 0 0 var(--s-xs) 0;
}

.nav-icon-bottom {
  float: right;
}

.nav-icon-top {
  float: left;
}

.nav-container {
  padding: var(--s-xs);
}

.nav-container-bottom {
  padding-right: var(--s-s);
  border-radius: var(--s-s) 0 0 0;
}

.nav-container-top {
  padding-right: var(--s-s);
  border-radius: 0 0 var(--s-s) 0;
}