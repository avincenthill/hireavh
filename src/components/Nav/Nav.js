import './Nav.css';
import { GoGrabber } from 'react-icons/go';
import { IconContext } from 'react-icons';
import Link from 'components/Link/Link';
import React from 'react';
import content from 'content/content';
import styleconfig from 'styles/styleconfig';

class Nav extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isExpanded: false,
    };
  }

  handleClick = (e) => {
    if (!this.state.isExpanded) {
      document.addEventListener('click', this.handleClick, false);
    } else {
      document.removeEventListener('click', this.handleClick, false);
    }
    this.toggleNav();
  };

  toggleNav = () => {
    this.setState({ isExpanded: !this.state.isExpanded });
  };

  renderNotExpanded = () => {
    return (
      <div
        className={`nav-icon $
          this.props.isBottom ? "nav-icon-bottom" : "nav-icon-top"
        }`}
      >
        <IconContext.Provider
          value={{
            style: { ...styleconfig.icons.l, ...styleconfig.icons.nav },
          }}
        >
          <GoGrabber onClick={this.handleClick} />
        </IconContext.Provider>
      </div>
    );
  };

  renderExpanded = () => {
    return (
      <div>
        <div
          className={`nav-container ${
            this.props.isBottom ? 'nav-container-bottom' : 'nav-container-top'
          }`}
        >
          <NavList></NavList>
        </div>
      </div>
    );
  };

  render() {
    return (
      <div className={`nav ${this.props.isBottom ? 'nav-bottom' : 'nav-top'}`}>
        {this.state.isExpanded
          ? this.renderExpanded()
          : this.renderNotExpanded()}
      </div>
    );
  }
}

class NavList extends React.Component {
  render() {
    return (
      <div className="nav-list">
        {Object.keys(content.nav.links).map((key, index) => {
          let link = content.nav.links[key];
          return (
            <Link
              key={index}
              title={link.emoji + ' ' + link.path}
              path={`/${link.path}`}
            ></Link>
          );
        })}
      </div>
    );
  }
}

export { Nav, NavList };
