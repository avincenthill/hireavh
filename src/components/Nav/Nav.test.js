import React from "react";
// import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Nav, NavList } from "./Nav";

describe("Nav", () => {
  describe("snapshot", () => {
    it("matches the last saved snapshot", () => {
      const tree = renderer.create(<Nav />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("NavList", () => {
  describe("snapshot", () => {
    it("matches the last saved snapshot", () => {
      const tree = renderer.create(<NavList />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
