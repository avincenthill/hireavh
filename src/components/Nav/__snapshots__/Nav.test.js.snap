// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot matches the last saved snapshot 1`] = `
<div
  className="nav nav-top"
>
  <div
    className="nav-icon $
          this.props.isBottom ? \\"nav-icon-bottom\\" : \\"nav-icon-top\\"
        }"
  >
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "var(--c-1)",
          "height": "3rem",
          "width": "3rem",
        }
      }
      viewBox="0 0 8 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 4v1H0V4h8zM0 8h8V7H0v1zm0 3h8v-1H0v1z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`NavList snapshot matches the last saved snapshot 1`] = `
<div
  className="nav-list"
>
  <div>
    <h2
      className="link"
    >
      <a
        className="link-a"
        href="/about"
      >
        <span>
          🏠 about
        </span>
      </a>
    </h2>
  </div>
  <div>
    <h2
      className="link"
    >
      <a
        className="link-a"
        href="/blog"
      >
        <span>
          ✍️ blog
        </span>
      </a>
    </h2>
  </div>
  <div>
    <h2
      className="link"
    >
      <a
        className="link-a"
        href="/projects"
      >
        <span>
          📦 projects
        </span>
      </a>
    </h2>
  </div>
  <div>
    <h2
      className="link"
    >
      <a
        className="link-a"
        href="/resume"
      >
        <span>
          📄 resume
        </span>
      </a>
    </h2>
  </div>
  <div>
    <h2
      className="link"
    >
      <a
        className="link-a"
        href="/contact"
      >
        <span>
          📇 contact
        </span>
      </a>
    </h2>
  </div>
</div>
`;
